Extend config to allow setting custom icon and highlight

ttytm opened the pull request:
The PR extends the config and adds type annotations. With the update it will be possible to easily customize the sign via the config. It can also be a base for further extensions.

```lua
require("bookmarks").setup({
	sign = { icon = "" },
})
```

LintaoAmons commented:
Hi Tt,

Thanks for making this PR, and I would love to add this function to the plugin.

I have some comments here, if you have some time can take a look, or I could do it later when I got some time~ 

1. I don't think we need make this name `BookmarksNvimSign` to be configurable.
2. Maybe makes the configuration more user understandable. Like:

```lua
{
  icon = "",
  icon_color = "red",
  line_font_color = "blue",
  line_bg_color = "red"
}
```

3. change the README accordingly

ttytm commented:
> 1. I don't think we need make this name `BookmarksNvimSign` to be configurable.
Good point about the `BookmarksNvimSign`, agreed .

> 2. Maybe makes the configuration more user understandable. Like:
> 
> ```lua
> {
>   icon = "",
>   icon_color = "red",
>   line_font_color = "blue",
>   line_bg_color = "red"
> }
> ```
Personally, I would keep the config for the sign logically structured in a table instead of using a prefix. When extending the config it would could get unappealing quickly.

E.g., a feature I thought of is that it would be cool to be able to have multiple signs. E.g. when setting a mark without a note, and when annotating something with a mark.

![2024-04-27_23-51](https://github.com/LintaoAmons/bookmarks.nvim/assets/34311583/fbba90d9-9cb9-40f6-9f26-16ab2a973583)

I updated the PR be easily extensible for such potentials

> 3. change the README accordingly

Updated, the push includes an auto-formatting change. Let me know if this should be reverted
